### ISSUE TITLE
Changing input size

### ISSUE THREAD

**Issue by off99555** [AlexeyAB/darknet#836]
I want to change tiny yolo v3 input size from 416x416 to 104x104 because I have small images and I want it to train faster. What parts of this file do I need to change?
Usually, in the yolo v2 model, I would just remove some max pooling layers and it would be fine. Now, this cfg file is complicated and I don't know where to start.
```
[net]
# Testing
batch=1
subdivisions=1
# Training
# batch=64
# subdivisions=2
width=416
height=416
```

**Comment by AlexeyAB** [comment]
* use `width=128 height=128` without removing layers (width and height should be multiple of 32): https://github.com/AlexeyAB/darknet/blob/e29fcb703f8d936e17507bf78043a8b8bc6279b0/cfg/yolov3-tiny.cfg#L8-L9

* also you can decrease `filters=512` or `256` in this layer: https://github.com/AlexeyAB/darknet/blob/e29fcb703f8d936e17507bf78043a8b8bc6279b0/cfg/yolov3-tiny.cfg#L99

* and you should fix it: https://github.com/AlexeyAB/darknet/blob/e29fcb703f8d936e17507bf78043a8b8bc6279b0/cfg/yolov3-tiny.cfg#L175
    to `mask = 0,1,2`

If you will remove downsampling layers such as max-pool layers with stride=2 - it will increase resulotion of the subsequent layers and it will be decrease performance.

**Comment by off99555** [comment]
What does the `mask` do?
Do I need to change it if I didn't change width and height to 128?

**Comment by AlexeyAB** [comment]
> What does the mask do? When do I need to change it?

There is a bug, it should be: `mask = 0,1,2` - this is indecies of anchors that will be used for this [yolo] layer

> [yolo]
> mask = 0,1,2
> anchors = **10,14,  23,27,  37,58**,  81,82,  135,169,  344,319

So will be used bolded anchors in this yolo-layer.

**Comment by off99555** [comment]
Yeah, so if it's a bug then it should be 0,1,2 always regardless of input size or model right?

**Comment by AlexeyAB** [comment]
@off99555 not so simple) File https://pjreddie.com/media/files/yolov3-tiny.weights trained for `mask = 1,2,3` so it will work incorrect with `mask = 0,1,2`. We are waiting for the new `yolov3-tiny.weights`

**Comment by off99555** [comment]
Ah, I see. So if I am training from scratch without any pre-trained weights, I should use `mask=0,1,2` but if I use the pre-trained weights provided by `darknet` as of now, I need to use `mask=1,2,3`. Am I right?

**Comment by AlexeyAB** [comment]
If you train your own network (regardless of with or without any pre-trained weights) then use `mask=0,1,2`
If you use exactly `yolov3-tiny.weights` and exactly for Detection, then use `mask=1,2,3`

**Comment by off99555** [comment]
Thank you. I have another error. It's when I train the model 128x128 to detect 133 classes.
Command:
`./darknet detector train data/ch.data ./cfg/yolov3-tiny-ch.cfg -dont_show`
Head of the output:
![2018-05-17_20-53-42](https://user-images.githubusercontent.com/15215732/40182130-ea9632a8-5a14-11e8-97b7-a2f9b64b8db0.png)
Tail of the output:
![2018-05-17_20-52-52](https://user-images.githubusercontent.com/15215732/40182117-e181c088-5a14-11e8-8669-71972554018e.png)

**Comment by AlexeyAB** [comment]
I fixed it. Update your code from this GitHub repository.

When you set `random=1` then for each 10 iterations the network will be resized randomly +-160 pixels. It allows to train such that it can detect object with different sizes.

**Comment by AlexeyAB** [comment]
Yes, you can set `random=0` in each yolo-layers.
